app/src/main/java/com/gacode/relaunchx/ReLaunch.java

```diff
@@ -1746,6 +1676,34 @@ public boolean onTouch(View v, MotionEvent event) {
 		refreshBottomInfo();
 	}
 
+	private int getFirstItemToScrollUp(GridView gv) {
+		int viewHeight = gv.getHeight();
+		int targetItem = gv.getFirstVisiblePosition();
+			while (targetItem >= 0 && viewHeight > itemsHeights.get(targetItem)) {
+				if(currentColsNum > 1)
+					viewHeight -= Collections.max(
+							itemsHeights.subList(targetItem, targetItem + currentColsNum));
+				else
+					viewHeight -= itemsHeights.get(targetItem);
+				targetItem -= currentColsNum;
+			}
+		return targetItem + currentColsNum;
+	}
+
+	private void setItemsHeights(GridView gv) {
+		int firstVisible = gv.getFirstVisiblePosition();
+		int lastVisible = gv.getLastVisiblePosition();
+		int visibleItems = lastVisible - firstVisible;
+		for(int i = 0; i <= visibleItems; i++) {
+            View item = gv.getChildAt(firstVisible);
```

Comment:
Did you mean getChildAt(firstVisible + i)

Reply:
yes :)